### ISSUE TITLE
MathML expressions in EPUBs are not readable with NVDA in some installations

### ISSUE THREAD

**Issue by ways2read** [edrlab/thorium-reader#1709]
On Windows 11 computer start NVDA 2022.1 

Launch Thorium Reader 2.0.0 and open the test EPUB [Accessibility Tests Mathematics](https://www.epubtest.org/books/Accessibility-Tests-Mathematics-v1.1.1.epub)

When encountering math expressions you just hear 'clickable' with either Access8Math version 3.3 or MathCAT 0.1.14 .

Go to the web page [Quadratic formula](https://en.wikipedia.org/wiki/Quadratic_formulal) page in Wikipedia and the math is readable and navigable with Access8Math and MathCAT as expected.

Reading math with this combination worked briefly on this computer before it stopped. I have uninstalled Thorium, NVDA, MathCAT and reinstalled but still broken. Math in Thorium with NVDA works for some people and not for others.

**Comment by ways2read** [comment]
Same machine, uninstall Thorium Reader (store version) and install [version 1.7.3](https://github.com/edrlab/thorium-reader/releases/tag/v1.7.3). Math is now read with MatchCAT.

**Comment by NSoiffer** [comment]
In some experiments today when math wasn't being read, NVDA reports "not math" when trying to navigate the math. This means NVDA didn't see this as math, which very likely means the hidden MathML is not there. In looking through the code, I see in `main\streamer\streamerCommon.ts` the line:
```
    // enableAssistiveMml: true,
```
`enableAssistiveMml` is what controls whether the hidden MathML is present or not. I know that in most MathJax configurations, this is enabled, but maybe somehow MathJax is getting into a state where it doesn't get enabled. 

Looking through the history in this file, this line was introduced in https://github.com/edrlab/thorium-reader/commit/910c3c59133d41926ee086c807029fbee5a7cac3. When introduced, it was commented out. Is there some reason it isn't set to true?

**Comment by ways2read** [comment]
I wondered if the screen reader was not being detected in some cases and MathJax was generating its own accessible MathML description (this messed up screen readers in #1602). However, that doesn't seem to be the case:
- start NVDA, open math test book in Thorium 1.7.3, math is read correctly with MathCAT, close Thorium
- open math test book in Thorium 2.0.0 math is not read with either MathCAT or Access8Math, reports "not math" with NVDA+shift+m
- quit NVDA, use Thorium read aloud, hear that it is using the authored alt text encoded in the MathML, so this has not been replaced by MathJax (i.e. Thorium seems to have detected the screen reader), quit Thorium, quit NVDA
- open math test book in Thorium 2.0.0 and use read aloud, now we hear different speech when on math expressions, a screen reader was not detected on launch and text text for read aloud has been generated by MathJax.

**Comment by danielweck** [comment]
Thank you Richard for this very helpful analysis (as discussed in private emails).

Neil, we tried various different tweaks in Thorium's MathJax configuration, but clearly we are still experiencing issues with some combinations of screen readers + optional math plugins. We also enable MathJax's dynamic generation of accessible text (speech utterance) which is helpful in cases where there is no authored alttext (ref. "speech rule engine" by @zorkow ).

Side note: this caught my attention in the MathJax issue tracker:

https://github.com/mathjax/MathJax/pull/2463/files

**Comment by NSoiffer** [comment]
The MathJax PR doesn't seem relevant. First off, it is for "legacy v2".
Thorium uses V3 I hope. Second it goes from what I guess is a bad value to
"false", so it turns it off. It does appear that V3 has the setting as true
as the default, so I doubt my suggestion solves the problem, but I don't
know why it would hurt.

**Comment by NSoiffer** [comment]
@danielweck: is there a way to view the DOM in thorium to verify whether the MathML is there or not?

**Comment by danielweck** [comment]
yes, debugging tools can be accessed with the automated builds downloaded from the GitHub releases, or with local dev builds.
in other words: production builds / official releases prevent access to the Chromium Web Inspector, Dev builds don't.

**Comment by danielweck** [comment]
Closing as not actionable, and since then MathJax vs. native Chromium MathML support has evolved.
